pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -607,8 +616,18 @@ func (c *ClusterConfig) validateKubernetesNetworkConfig() error {
 			if missing := c.addonContainsManagedAddons([]string{VPCCNIAddon, CoreDNSAddon, KubeProxyAddon}); len(missing) != 0 {
 				return fmt.Errorf("the default core addons must be defined for IPv6; missing addon(s): %s; either define them or use EKS Auto Mode", strings.Join(missing, ", "))
 			}
-			if c.IAM == nil || c.IAM != nil && IsDisabled(c.IAM.WithOIDC) {
-				return fmt.Errorf("oidc needs to be enabled if IPv6 is set; either set it or use EKS Auto Mode")
+			if len(c.addonContainsManagedAddons([]string{PodIdentityAgentAddon})) != 0 && (c.IAM == nil || c.IAM != nil && IsDisabled(c.IAM.WithOIDC)) {
+				return fmt.Errorf("either pod identity or oidc needs to be enabled if IPv6 is set; set either one or use EKS Auto Mode")
+			}
+
+			if len(c.addonContainsManagedAddons([]string{PodIdentityAgentAddon})) == 0 && !c.AddonsConfig.AutoApplyPodIdentityAssociations {
+				// Assuming user intends to use pod identities if the pod identity agent addon is added.
+				vpcCNIAddonEntry := c.getAddon(VPCCNIAddon)
+
+				if !vpcCNIAddonEntry.UseDefaultPodIdentityAssociations &&
```

Comment:
now that `getAddon` is a function that can return nil, you need to do a nil check first here.

Reply:
`getAddon(VPCCNIAddon)` should never return `nil` because
```go
missing := c.addonContainsManagedAddons([]string{VPCCNIAddon, CoreDNSAddon, KubeProxyAddon}); len(missing) != 0
```

is `false` on L616. Do you still want me to add a `nil` check given that?

Reply:
Yeah I think it would be good, otherwise if we modify this code in the future it may be easy to miss.